idx/idx.go

```diff
@@ -99,39 +55,41 @@ type MetricIndex interface {
 	// Stop shuts down the index.
 	Stop()
 
+	// UpdateMaybe updates an existing archive, if found.
+	// it returns the existing archive (if any), and whether it was found
+	UpdateMaybe(point schema.MetricPoint, partition int32) (Archive, int32, bool)
```

Comment:
can we just call this `Update()`   updateMaybe sounds weird to me.  It makes it sounds like the update may or may not happen due to some unknown reason.

This is just a simple update request that will either succeed or fail due to there being being no existing object to update.

Reply:
+1 I'm not a fan of the BlahMaybe naming in general.

Reply:
+1 i like it